### ISSUE TITLE
Multiple ideas for future features

### ISSUE THREAD

**Issue by D3nn3** [morpheus65535/bazarr#128]
Hi there,

I'm using Bazarr not to long (just couple days) and it works great so far. It's stable and does it's job. Thank you for that! But I would love to see some improvements:

1. A more fine grained way to chose which subtitles to download. At the moment we only can chose a score (in steps of 5) which isn't enough imho. I'm more the type of guy who wants to have more control. Some ideas:
1.1 Provider score boost or something like preferred providers. 
1.2 Choose which fields (like title, series, season, episode, **format**, **release group**, ...) have to match etc. At the moment I don't like that subliminal uses filename guessing, gets a score of >95% and it's wrong way to often. Sub-Zero (Plex plugin, uses subliminal as well) for example uses a better scoring system which gives me more control: https://github.com/pannal/Sub-Zero.bundle/wiki/Media-Score. It's also not perfect but It works really well. You could probably just copy it.

2. Mechanism to upgrade to a better subtitle (like Sonarr does with TV Shows). Once a subtitle is downloaded it's the end of the story. A better subtitle comes out? No way that Bazarr automatically downloads the better (= higher score) version. Bazarr could check for X days for new episodes if a better subtitles comes along. After these x days Bazarr won't check for this episode anymore.

3. Display a list of available subtiles and let me chose which one to download (like Sonarr)

4. Link to the subtitles which got downloaded (for example in the history and see point 5).

5. Display subtitles and their relevant information on the series page. The history is nice but with over 2800 subtitles download in the past couple days I won't even try to search for items. @ relevant information: At the moment I just see the score and provider of the subtitle which was downloaded. Some more information like the link to the provider page, format, release_group, etc. would be nice.

Just some ideas :) I would love to replace Sub-Zero with Bazarr but at the moment some key features (automatically upgrade subtitles + better scoring system) are missing :(

**Comment by morpheus65535** [comment]
Thanks for your suggestion.

1. The only thing that I see that Sub-Zero have that Bazarr don't is the addic7ed-boost. Regarding the score, in fact, you have more control with 20 possible values vs 4 in Sub-Zero.
2. Could be a good feature to add in the future but would require some major change. I'll keep this one in mind.
3. I'm working on a manual search functionality at the moment and it should be in version 0.7.0.
4. (and 5) This could be added to history. I'm planning an "history" view by episode / movie like in Sonarr and Radarr. Maybe in 0.7.0.

I know that some user like very fine grained settings but I don't want Bazarr to become too complicated fore average user just to please some user. I have to keep just enough settings.

**Comment by D3nn3** [comment]
Hi, thanks for the reply.

1. You don't have only 4 possible values, you can input any number between 0 and 359 for tv shows and 0 - 119 for movies. I'm using 345 for tv shows (series 180 + season 30 + episode 30 + year 90 + release_group 15). I chose this value specifically to force a matching release_group. This way I rarely/never get wrong subtitles. 

2. I don't know. You would basically need a database to store each episode, day of import, the ID from the subtitles and the corresponding score if a subtitle already exists. This database already exists, don't know if all the info is in there already, didn't check. Create a task which periodically searches for better better subtitles of each episode which isn't older than X days and the score is below a threshold . That's (more or less) it.

3. Glad to hear! 

4. Good news.

@ fine grained settings: Many programs solve this by using advanced options for users who like them.

**Comment by D3nn3** [comment]
1. How about this: Just let the users change the percentage value by steps of 1.

E.g. : 
series 180 + season 30 + episode 30 + year 90 + format 7 = 337 = 93,6%
series 180 + season 30 + episode 30 + year 90 + release_group 15 = 345 = 95,8%
series 180 + season 30 + episode 30 + year 90 + release_group 15 + format 7 = 352 = 97,7%

At the moment a percentage of 90% would match anything from 324 and above which would give you a lot of wrong subtitles. 95% seems to be reasonable at the moment. But people wanting 93% (or 97%) and above don't have this option at the moment. Being able to change the percentage by steps of 1 would be enough for the moment I guess.

Other criterias like video_codec and resolution have such a low score that it shouldn't mess anything up in most cases. 

2. I'll trust you on this one :D Didn't think that it would hold every data that would ne necessary for such a task.

**Comment by hotio** [comment]
I agree that a better matching system should be used, at the very least make (format, release group) a required option for users that don't want false positives. Like subzero is also good...I agree with @D3nn3

**Comment by morpheus65535** [comment]
@D3nn3 Beginning with version 0.7.1 (in dev now), user is going to be able to increment minimum score by 1. This is a temporary solution and we should get something better in the future.

**Comment by morpheus65535** [comment]
@D3nn3 Starting with 0.7.3 in dev branch actually, Bazarr update subtitles.

**Comment by D3nn3** [comment]
That's great to hear, thank you very much!
Sent you 5$ for a beer as a token of my appreciation :) Keep up the good work

**Comment by morpheus65535** [comment]
Thanks!
